portfolio/src/test/java/com/google/sps/DatastoreUtilsTest.java

```diff
@@ -188,4 +188,11 @@ public void PutBlogsInDatastoreTest_withoutPost() {
     assertEquals(0, ds.prepare(new Query(BlogConstants.BLOG_ENTITY_KIND)).countEntities(withLimit(5)));
   }
 
+  @Test
+  public void testDoGetFromDatastore() {
+    createDatastoreEntities();
+    LinkedList<BlogMessage> actual = DatastoreUtils.doGetFromDatastore(1);
+
+    Assert.assertEquals(1, actual.size());
```

Comment:
You need some extra checks here. You want this to fail if doGetFromDatastore changes in an unwanted way. doGetFromDatastore has 3 functions that it calls: putRepliesWithPosts, updateTagsToSearch, and sortAndLoadfromBlogHashmap. You will need to check the output from doGetFromDatastore to confirm that each of these are called.